streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -227,13 +289,70 @@
                                        final Serde<K> keySerde,
                                        final Serde<V> valSerde,
                                        final Pattern topicPattern) {
+        return stream(offsetReset, null, keySerde, valSerde, topicPattern);
+    }
+
+    /**
+     * Create a {@link KStream} from the specified topic pattern.
+     * <p>
+     * If multiple topics are matched by the specified pattern, the created {@link KStream} will read data from all of
+     * them and there is no ordering guarantee between records from different topics.
+     * <p>
+     * Note that the specified input topics must be partitioned by key.
+     * If this is not the case it is the user's responsibility to repartition the date before any key based operation
+     * (like aggregation or join) is applied to the returned {@link KStream}.
+     *
+     * @param timestampExtractor the timestamp extractor used for this source {@link KStream},
+     *                           if not specified the default extractor defined in the configs will be used.
+     * @param keySerde           key serde used to read this source {@link KStream},
+     *                           if not specified the default serde defined in the configs will be used
+     * @param valSerde           value serde used to read this source {@link KStream},
+     *                           if not specified the default serde defined in the configs will be used
+     * @param topicPattern       the pattern to match for topic names
+     * @return a {@link KStream} for topics matching the regex pattern.
+     */
+    public <K, V> KStream<K, V> stream(final TimestampExtractor timestampExtractor,
```

Comment:
Nit: Can you insert this method further down -- we want to order method overloads with regard to number of parameters -- it simplifies to keep track of what overloads are there.